### PR TITLE
Authorization plugin major refactoring

### PR DESCRIPTION
**Pull request by IlliaYalovoi**
Completely deleted kafka from gateway, remake all with sync http calls. I think kafka, and all complex concurrency, is a main reason of nginx timeout error. I deployed the project on dev server (http://195.22.131.17:31082) with this janus version Jan 2 10:00, and I think if it will work 24+ hours without errors, this PR can be safely merged

Instead of router, I use 'catching' middleware. This meant to be used with 'technical' endpoints (gatewayTokens, gatewayRoles) which doesn't redirect anywhere, but catcher catches it, and respond with 200 status code. Security is not a problem, I think, because gateway port must be closed from external connection, and only accessible from frontend

### CODE REVIEW

File: pkg/plugin/authorization/http.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	"time"
+)
+
+var (
+	ErrTimeout = errors.New("timed out")
+)
+
+// doGetRequestWithTimeout sends GET http request on passed URL with 3 sec timeout.
+func doGetRequestWithTimeout(url string) ([]byte, error) {
+	client := http.Client{Timeout: 3 * time.Second}
```
Comment (Bykwarb) [inline pkg/plugin/authorization/http.go]: I think timeout in this func should be an argument. Something like this:
```
doGetRequestWithTimeout(url string, timeoutDuration int) ([]byte, error)
```

**Review by Bykwarb — APPROVED** [review]

File: pkg/plugin/authorization/http.go (hunk 1 above)
Comment (IlliaYalovoi) [reply pkg/plugin/authorization/http.go]: Nice adjustment, thanks

File: pkg/plugin/authorization/middleware.go [hunk 2]
```diff
@@ -80,54 +94,154 @@ func NewRoleCheckerMiddleware() func(http.Handler) http.Handler {
...
 	return true
 }
+
+func NewTokenCatcherMiddleware(manager *TokenManager) func(http.Handler) http.Handler {
+	return func(handler http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			fmt.Println(r.URL.Path, r.URL.Path == fmt.Sprintf("/%s/gatewayTokens", manager.Conf.ApiVersion))
+			switch r.URL.Path { // path doesnt case
+			case fmt.Sprintf("/%s/gatewayTokens", manager.Conf.ApiVersion):
+				fmt.Println(1)
```
Comment (Bykwarb) [inline pkg/plugin/authorization/middleware.go]: unnecessary log

File: pkg/plugin/authorization/middleware.go [hunk 3]
```diff
@@ -80,54 +94,154 @@ func NewRoleCheckerMiddleware() func(http.Handler) http.Handler {
...
+					errors.Handler(w, r, ErrUnmarshal)
+				}
+
+				manager.UpsertTokens(tokens)
+
+				render.JSON(w, http.StatusOK, http.NoBody)
+				return
+
+			case fmt.Sprintf("/%s/gatewayTokens/delete", manager.Conf.ApiVersion):
+				fmt.Println(2)
```
Comment (Bykwarb) [inline pkg/plugin/authorization/middleware.go]: same

File: pkg/plugin/authorization/middleware.go [hunk 4]
```diff
@@ -80,54 +94,154 @@ func NewRoleCheckerMiddleware() func(http.Handler) http.Handler {
...
 		}
 	}
 
 	return true
 }
+
+func NewTokenCatcherMiddleware(manager *TokenManager) func(http.Handler) http.Handler {
+	return func(handler http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			fmt.Println(r.URL.Path, r.URL.Path == fmt.Sprintf("/%s/gatewayTokens", manager.Conf.ApiVersion))
```
Comment (Bykwarb) [inline pkg/plugin/authorization/middleware.go]: probably too

Comment (IlliaYalovoi) [reply pkg/plugin/authorization/middleware.go]: Fixed

File: pkg/plugin/authorization/middleware.go (hunk 3 above)
Comment (IlliaYalovoi) [reply pkg/plugin/authorization/middleware.go]: Fixed

File: pkg/plugin/authorization/middleware.go (hunk 2 above)
Comment (IlliaYalovoi) [reply pkg/plugin/authorization/middleware.go]: Fixed

**Pull request merged**